imxrt-hal/Cargo.toml

```diff
@@ -26,15 +26,23 @@ test = false
 
 [features]
 default = ["embedded-hal/unproven"] # Allows us to access the new digital pin traits
+
+# Device level features to allow for simpler conditional compiles
+imxrt101x = []
+imxrt102x = []
+imxrt105x = []
+imxrt106x = []
```

Comment:
Will these be something others will depend on, and we'll need to support going forward? Or, are these just for our internal use?

If it's the latter, we should document that these are not for crate users. Maybe we prefix them with a `_`?

We could try something like the tokio team, where they have [internal configuration macros](https://github.com/tokio-rs/tokio/blob/master/tokio/src/macros/cfg.rs) to express combinations of features. The macros are [used to selectively enable modules](https://github.com/tokio-rs/tokio/blob/master/tokio/src/lib.rs#L303), or functions, or structs. With this approach, all feature-flags are usable by users, so there's no ambiguity. And, we can replace things like

```rust
#[cfg(any(feature = "imxrt1060", feature = "imxrt1062", feature = "imxrt1064"))]
pub mod foobar { /* ... */ }
```

with

```rust
cfg_imxrt106x! {
    pub mod foobar { /* ... */ }
}
```

Reply:
The macro wrapper is a neat idea, I'll take a closer look at that. That's the general idea though, to internally allow maximum code sharing without typing massive #[cfg()] options